src/server/frontend_xwayland/xwayland_surface_observer.cpp

```diff
@@ -113,6 +113,12 @@ void mf::XWaylandSurfaceObserver::input_consumed(ms::Surface const*, MirEvent co
 {
     if (mir_event_get_type(event) == mir_event_type_input)
     {
+        // Ignore key repeat events
+        auto const input_event = mir_event_get_input_event(event);
+        if ((mir_input_event_get_type(input_event) == mir_input_event_type_key) &&
+            (mir_keyboard_event_action(mir_input_event_get_keyboard_event(input_event)) == mir_keyboard_action_repeat))
+                return;
```

Comment:
I don't believe this is necessary. If it doesn't return here all it does is call `input_dispatcher->handle_event()` from the Wayland thread, `WaylandInputDispatcher::handle_event()` calls `WaylandInputDispatcher::handle_keyboard_event()` and that does nothing unless the action is `mir_keyboard_action_down` or `mir_keyboard_action_up`. Aside from negligibly improve performance, this logic would seem to have no effect. If it *did* do anything important, we would need to implement that in the `WaylandSurfaceObserver` as well.